cmd/queue/main.go

```diff
@@ -298,8 +257,19 @@ func main() {
 		logger.Fatal("Error creating new config", zap.Error(err))
 	}
 	kubeClient = kc
-	go connectStatSink()
+
+	go func() {
+		autoscalerEndpoint := fmt.Sprintf("ws://%s.%s.svc.cluster.local:%s", servingAutoscaler, system.Namespace, servingAutoscalerPort)
+		logger.Infof("Connecting to autoscaler at %s", autoscalerEndpoint)
+		conn, err := websocket.NewDurableSendingConnection(autoscalerEndpoint)
```

Comment:
You have to put this in a goroutine because it's blocking.  But you want to make it non-blocking for pod startup.

How about just doing the connection stuff asynchronously in the library?  Making this a non-blocking call.  Then you can pull this out of the extra goroutine.

Reply:
Do we really want to make this blocking for pod startup? That'd mean the pod startup latency depends on how quick we can establish this connection. I rather liked that being entirely asynchronous.

Reply:
Ah, I think I see what you mean: I could guard the internal methods (Send and Close) against not having a proper connection and then make this non-blocking? Is that what you're saying?

Reply:
Yeah, in `main.go` you can keep the Knative Websocket object.  And either implement an `IsConnected()` function or implement a `MaybeSend()` method that drops messages when not connected.

That way you can also control concurrent access to the `*websocket.Connection` object, which we are not currently doing !?